codec/codec.go

```diff
@@ -0,0 +1,46 @@
+package codec
```

Comment:
I did not mean the transport package needs to know codec. I meant where to record the codec to be used. In the current code, if you want to record the codec in the grpc package, you need to deal with 4 cases differently -- (unary|streaming rpc) x (client|server). If the codec is recorded in the transport (actually transport.Stream struct which is common for all the above 4 cases), it is simplified. But I do not want users to interact with a private package (the transport package) directly. So I added a new package to do that.